### PR TITLE
Add ARM endpoint, update model responses/inputs to align with expected values

### PR DESCRIPTION
**Pull request by r-delgadillo**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by r-delgadillo** [comment]
Add @emgarten @zzcodework @amrenbin @cle5eland

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json [hunk 1]
```diff
@@ -338,20 +338,66 @@
...
           {
-            "name": "operationInputs",
+            "name": "nameAvailabilityInputs",
             "in": "body",
-            "description": "Set the name parameter in the OperationInputs structure to the name of the IoT Central application to check.",
+            "description": "Set the name parameter in the NameAvailabilityInputs structure to the name of the IoT Central application to check.",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/OperationInputs"
+              "$ref": "#/definitions/NameAvailabilityInputs"
```
Comment (jianghaolu) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: Is this change of name necessary? This is a breaking change on the client side.

Comment (r-delgadillo) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: Would it be better if I reverted the name back to `OperationInputs` and add the `subdomain` property to this model. The only problem I encountered when doing this was having set the `required` field to false for the `req.body`. Not sure if this is the best approach, thoughts?

Comment (jianghaolu) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: @r-delgadillo Adding a property that's not required is okay but changing a required property to optional is a breaking change.

Comment (jianghaolu) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: Why do you need to set the body parameter to be optional? Seems like the difference between `OperationInputs` and `NameAvailabilityInputs` is just an optional property `type`.

Comment (r-delgadillo) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: @jianghaolu The reason for this change was to make it clear to the user that the two endpoints `checkNameAvailability` and `checkSubdomainAvailability` have different input parameters, one with `name` and the other `subdomain`. After discussing with the team, we decided to keep `OperationInputs` and pass this model into both endpoints `req.body`. Thanks for the feedback.

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json [hunk 2]
```diff
@@ -537,22 +583,37 @@
     },
     "ErrorDetails": {
       "description": "Error details.",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/ErrorResponseBody"
```
Comment (jianghaolu) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: Please add `x-ms-flatten: true` to reduce breaking change from the client side

Comment (jianghaolu) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: See https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-flatten for documentation.

Comment (r-delgadillo) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json]: xxsx

**Comment by jianghaolu** [comment]
https://github.com/Azure/azure-rest-api-specs/pull/4015/files#diff-23bf8d6e84c93efe6c0e1b736fb4a61cR716 is a breaking change but the original spec was incorrectly describing the API. Merge as a fix.

**Review by jianghaolu — APPROVED** [review]

**Review by amrenbin — APPROVED** [review]

**Review by KrisBash — APPROVED** [review]
Signing off from ARM

**Comment by jhendrixMSFT** [comment]
@AutorestCI rebase azure-sdk-for-go

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**